spec/rspec/mocks/any_instance_spec.rb

```diff
@@ -1213,6 +1230,17 @@ def dup(_)
         end
       end
 
+      context "on a grandchild class" do
+        it "stubs a method not implemented in grandchild" do
+          allow_any_instance_of(::AnyInstanceSpec::ChildClass).to receive(:foo).and_return :result
+          expect(::AnyInstanceSpec::ChildClass.new.foo).to eq :result
```

Comment:
These tests would not fail with anonymous classes!  I had to pollute the test suite with some test constants so I wrapped them in a module to avoid a collision.  I'm open to doing this differently.

I originally had something like this but it wouldn't reproduce the ArgumentError.

```diff
diff --git a/spec/rspec/mocks/any_instance_spec.rb b/spec/rspec/mocks/any_instance_spec.rb
index f2375cf..e33b8b1 100644
--- a/spec/rspec/mocks/any_instance_spec.rb
+++ b/spec/rspec/mocks/any_instance_spec.rb
@@ -31,6 +31,12 @@ module RSpec
           def private_method; :private_method_return_value; end
         end
       end
+
+      # TODO: Remove me... anonymous classes can't reproduce #1042
+      let(:grandparent_class) { Class.new                    { def foo(_a); 'bar'; end }}
+      let(:parent_class)      { Class.new(grandparent_class) { def foo; super(:a); end }}
+      let(:child_class)       { Class.new(parent_class)      {}}
+
       let(:existing_method_return_value) { :existing_method_return_value }
```